cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/PullDocIdsFilter.java

```diff
@@ -0,0 +1,42 @@
+package com.cloudant.sync.replication;
+
+import com.cloudant.sync.internal.util.JSONUtils;
+import com.cloudant.sync.internal.util.Misc;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * <p>Sets the list of doc IDs to use as filtering criteria when a pull replication calls the
+ * source database's {@code _changes} feed.
+ * </p>
+ * Note: Doc IDs filtering are supported only when replicating against a CouchDB 2.x compliant database
+ *
+ * @see
+ * <a target="_blank" href="https://console.bluemix.net/docs/services/Cloudant/api/database.html#get-changes">See doc_ids filtering</a>
+ */
+public class PullDocIdsFilter extends PullFilter {
+
+    public static final String DOC_IDS_FILTER_NAME = "_doc_ids";
+
+    public PullDocIdsFilter(List<String> docIds) {
+        super(DOC_IDS_FILTER_NAME);
+
+        Misc.checkState(docIds != null && !docIds.isEmpty(),
+                "Doc Ids should not be empty in a Doc Ids Filter");
+
+        List<String> internalDocIds = new ArrayList<String>(docIds);
+        Collections.sort(internalDocIds);
+
+        Map<String, Object> docIdsParameter = new HashMap();
+        docIdsParameter.put("doc_ids", internalDocIds);
+
+        Map<String, String> parameters = new HashMap();
+        parameters.put("body", JSONUtils.serializeAsString(docIdsParameter));
```

Comment:
I don't really like this idea of unpacking and packing things via the "body" entry in the map. We can just use the map directly when doing the POST and get rid of this extra level of indirection.

Reply:
Actually looking at the original definition this was probably done to avoid encoding issues when sending the query string. Since we are now using the JSON encoder to send these parameters as in a POST request, <String,Object> is the right choice. I wonder if we can keep a <String,String> setter so that we don't break backwards compatibility?